lib/cli/cli.js

```diff
@@ -18,27 +18,114 @@ function CLI(options) {
 module.exports = CLI;
 
 CLI.prototype.run = function(environment) {
-  return Promise.hash(environment).then(function(environment) {
-    var args = environment.cliArgs.slice();
-    var commandName = args.shift();
-    var commandArgs = args;
+  return Promise.hash(environment)
+    .then(function(environment) {
```

Comment:
It becomes harder to review the changes because you're changing the syntnax here (moving `then` to a new line and indenting the block). I'd put that kind of stuff in a separate pr.